org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/navigation/FindImplementersHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.util.MethodOverrideTester;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandler;
+import org.eclipse.jdt.ls.core.internal.hover.JavaElementLabels;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EitherTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EnumTypeAdapterFactory;
+
+/**
+ * Command to find implementers of a type or a method
+ *
+ * @author dbocharo
+ */
+public class FindImplementersHandler {
+  private static final Gson gson =
+      new GsonBuilder()
+          .registerTypeAdapterFactory(new CollectionTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EitherTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EnumTypeAdapterFactory())
+          .create();
+
+  /**
+   * Finds implementers for an element defined by position
+   *
+   * @param parameters first parameter must be of type {@link TextDocumentPositionParams}, which
+   *     defines a position of an element which search implementers for
+   * @param pm a progress monitor
+   * @return an object of type {@link ImplementersResponse}, which contains description of a
+   *     searched element and implementers
+   */
+  @SuppressWarnings("restriction")
+  public static Object getImplementers(List<Object> parameters, IProgressMonitor pm) {
+    TextDocumentPositionParams param =
+        gson.fromJson(gson.toJson(parameters.get(0)), TextDocumentPositionParams.class);
+
+    ITypeRoot typeRoot = JDTUtils.resolveTypeRoot(param.getTextDocument().getUri());
+
+    ImplementersResponse implementersResponse = new ImplementersResponse();
+    List<SymbolInformation> implementers = new ArrayList<>();
+    implementersResponse.setImplementers(implementers);
+    try {
+      IJavaElement elementToSearch =
+          JDTUtils.findElementAtSelection(
+              typeRoot,
+              param.getPosition().getLine(),
+              param.getPosition().getCharacter(),
+              JavaLanguageServerPlugin.getPreferencesManager(),
+              pm);
+
+      switch (elementToSearch.getElementType()) {
+        case IJavaElement.TYPE:
+          findSubTypes(elementToSearch, implementers, pm);
+          implementersResponse.setSearchedElement(elementToSearch.getElementName());
+          break;
+        case IJavaElement.METHOD:
+          findTypesWithSubMethods(elementToSearch, implementers, pm);
+          implementersResponse.setSearchedElement(elementToSearch.getElementName());
+          break;
+        default:
```

Comment:
I dont' think we need the default clause.

Reply:
it's a matter of code style: https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck.

Reply:
Yeah, but you're not doing anything in the default clause. I would think using if-else if would more precisely express what you're doing here: we're not trying to cover all cases of element types.